q2_types/feature_data/_format.py

```diff
@@ -131,23 +136,82 @@ def sniff(self):
+                                f'{line_number-1!r}')
+                        last_line_was_ID = True
+                    elif re.fullmatch(FASTADNAValidator, line):
+                        last_line_was_ID = False
+                    else:
+                        raise ValidationError(
+                            f'Invalid sequence on line {line_number}')
+                    line_number += 1
+                    line = fh.readline()
+            except UnicodeDecodeError as e:
+                # We tell() on the buffer because we want the actual buffer the
+                # error occured in, tell() on the file handle will report being
+                # at the end of the last buffer when the error occured well
+                # into the next buffer
+                buffer = fh.buffer
+                pos = buffer.tell()
+                # We want to start our read from the beginning of the buffer we
+                # encountered the error on because we have yet to count any
+                # lines in that buffer
+                pos = (math.ceil(pos / io.DEFAULT_BUFFER_SIZE) - 1) * \
+                    io.DEFAULT_BUFFER_SIZE
+                buffer.seek(pos)
+                # e.start reports the position of the bad byte in the current
+                # buffer not the file, this is the cause of all the pain
+                # determining which buffer in the file the error occured in so
+                # we can find the bad byte's position in the file not just the
+                # buffer. If the bad byte is the final byte in a buffer it will
+                # report it as being the first byte in the next buffer. This is
+                # likely due to the fact that e.start reports the buffer
+                # position of the bad byte, and e.end reports the buffer
+                # position of the next good byte. If the last byte in a buffer
+                # is bad then e.start should be 8192 and e.end should be byte 0
+                # of the next buffer, but end being less than start doesn't
+                # really make sense, so this is handled by bumping e.start to 0
+                # and e.end to 1
+                fh_error = io.TextIOWrapper(io.BytesIO(buffer.read(e.start)),
+                                            errors='ignore')
+                # We need to count the lines in the buffer the error occured in
+                # only counting newline terminated lines to ensure we do not
+                # double count the line the error occured on, the line count is
+                # initialized to one which accounts for the error line, and the
+                # error line will always terminate with the bad byte not a
+                # newline
+                for line in fh_error:
+                    if line[-1] == '\n':
+                        line_number += 1
+                raise ValidationError('Unicode cannot decode byte on line '
```

Comment:
I think Unicode is the wrong term here as utf8 is the encoding (it might be a Unicode character of the wrong encoding for example)

Reply:
I do believe you are correct